spec/routes_spec.rb

```diff
@@ -74,14 +74,18 @@ def fp_message
               allow(subject).to receive(:redirect_to).with(:controller => 'dashboard', :action => 'auth_error').and_raise(MiqException::RbacPrivilegeException)
               allow(subject).to receive(:add_flash).with("The user is not authorized for this task or item.", :error).and_raise(MiqException::RbacPrivilegeException)
 
-              expect do
-                subject.send(:check_privileges)
-                subject.send(action)
-              rescue MiqException::RbacPrivilegeException
+              begin
+                expect do
+                  subject.send(:check_privileges)
+                  subject.send(action)
+                rescue MiqException::RbacPrivilegeException
+                  raise
+                end.to raise_error(MiqException::RbacPrivilegeException)
+              rescue RSpec::Expectations::ExpectationNotMetError
+                # Append a custom error message
+                $!.message << "\n\n" + error_message unless fp
```

Comment:
This hackery allows us to append to the normal `expect {}.to raise_error` error with more information regarding how to fix locations that aren't correctly raising the expected RBAC Privilege exceeption.

Reply:
Since this is difficult to visualize... here's an example of one that would previously fail to raise the correct rbac exception but instead raise our new missing feature one:

```
  1) Application routes MiqPolicyController#miq_policy_edit RBAC check is enforced
     Failure/Error:
       expect do
         subject.send(:check_privileges)
         subject.send(action)
       rescue MiqException::RbacPrivilegeException
         raise
       end.to raise_error(MiqException::RbacPrivilegeException)

       expected MiqException::RbacPrivilegeException, got #<RuntimeError: role_allows? with non-existing product feature identifier: ["policy_new"] is not supp...nversion_host_show_list", "conversion_host_show", "conversion_host_control", "conversion_host_tag"]> with backtrace:
         # ./app/helpers/application_helper.rb:117:in `role_allows?'
         # ./app/controllers/application_controller.rb:1835:in `block in assert_privileges'
         # ./app/controllers/application_controller.rb:1835:in `none?'
         # ./app/controllers/application_controller.rb:1835:in `assert_privileges'
         # ./app/controllers/miq_policy_controller/policies.rb:17:in `policy_edit_reset'
         # ./app/controllers/miq_policy_controller/policies.rb:94:in `miq_policy_edit'
         # ./spec/routes_spec.rb:81:in `block (8 levels) in <top (required)>'
         # ./spec/routes_spec.rb:79:in `block (7 levels) in <top (required)>'
         # /Users/joerafaniello/.gem/ruby/2.6.6/gems/webmock-3.12.2/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

       This route probably does not have an RBAC check enforced on itself. This is could be because of:
       * a missing feature that should be defined in the "#{controller}_#{action}" format
       * a missing `feature_for_actions` mapping in the controller between the action and a feature
       * missing explicit call(s) to assert_privileges in the action's method in the controller
       * the route is not available without a session variable that is being set in an enforced route (FP)
       * the route is a redirect to another route that has RBAC enforcement (FP)
       * the route is checked properly, but the test is somehow not detecting it (FP)
       * the route should be exposed to any logged in user without a role check (FP)

       The routes marked with (FP) should be treated as false positives and can be added into the
       `spec/config/routes.pending.yml` file. Please note that if the action is not available under any
       other controller, there is a high chance that it is not a false positive and it should not be
       added into this file without an explanation.
     # ./spec/routes_spec.rb:79:in `block (7 levels) in <top (required)>'
     # /Users/joerafaniello/.gem/ruby/2.6.6/gems/webmock-3.12.2/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
```